spec/controllers/exercises_controller_spec.rb

```diff
@@ -1,92 +1,212 @@
+        end
+      end
     end
 
-    it 'raises an active record error if the exercise does not exist' do
-      expect { get :show, id: 'invalid ID' }.to raise_error(ActiveRecord::RecordNotFound)
+    context 'As a guest' do
+      it 'redirects you to the sign-in / sign-up page' do
+        get :index
+        expect(response).to redirect_to '/users/sign_in'
+      end
+
+      it 'returns a 302 (redirect) response' do
+        get :index
+        expect(response).to have_http_status '302'
+      end
     end
+  end
+
+  describe '#show' do
+    let(:exercise) { FactoryBot.create(:exercise, user_id: current_user.id) }
+    let(:current_user) { FactoryBot.create(:user) }
+
+    context 'when logged in' do
+      before do
+        allow(controller).to receive(:authenticate_user!)
+        allow(controller).to receive(:current_user).and_return(current_user)
+      end
+
+      it 'responds successfully to a GET request' do
+        get :show, params: { id: exercise.id }
+        expect(response).to be_successful
+      end
 
-    it 'assigns the return value of exercise#rehearsals_for_user to the rehearsals ivar' do
-      exercise = FactoryBot.create(:exercise)
-      allow(Exercise).to receive(:find).with(exercise.id.to_s).and_return(exercise)
+      it 'returns a 200 HTTP response code' do
+        get :show, params: { id: exercise.id }
+        expect(response).to have_http_status('200')
+      end
 
-      rehearsals_for_user = double('rehearsals_for_user')
-      allow(exercise).to receive(:rehearsals_for_user).with(user).and_return(rehearsals_for_user)
+      it 'raises an active record error if the exercise does not exist' do
+        expect { get :show, params: { id: '2384' } }.to raise_error(ActiveRecord::RecordNotFound)
+      end
 
-      get :show, id: exercise.id
-      rehearsals_ivar = assigns(:rehearsals)
+      it 'blocks you (redirects you to root) from viewing a private, non-published/non-global exercise that does not belong to you' do
+        not_my_exercise = FactoryBot.create(:exercise, user_id: 35, global: false)
+        get :show, params: { id: not_my_exercise.id }
+        expect(response).to redirect_to root_path
+      end
+    end
+
+    context 'when not logged-in' do
+      it 'returns a 302 response' do
+        get :show, params: { id: exercise.id }
+        expect(response).to have_http_status('302')
+      end
 
-      expect(rehearsals_ivar).to eq(rehearsals_for_user)
+      it 'redirects to the sign-in page' do
+        get :show, params: { id: exercise.id }
+        expect(response).to redirect_to('/users/sign_in')
+      end
     end
+  end
 
-    context 'when the exercise has no tags' do
-      it 'doctrines ivar is nil' do
-        exercise = FactoryBot.create(:exercise)
+  describe '#edit' do
+    context 'As a signed-in, non-admin user' do
+      let(:current_user) { FactoryBot.create(:user) }
+      let(:exercise) { FactoryBot.create(:exercise, user_id: current_user.id) }
```

Comment:
Why not use the less verbose and more readable notation:
```ruby
FactoryBot.create(:exercise, user: current_user)
```
You may also want to add this to your `spec_helper.rb` or `rails_helper.rb`:
```ruby
RSpec.configure do |config|
  config.include FactoryBot::Syntax::Methods
end
```
You should then be able to use the DSL like this:
```ruby
 let(:current_user) { create(:user) }
 let(:exercise) { create(:exercise, user: current_user) }
```

Reply:
Yes I added the `FactoryBot` `config` in a previous commit.

I am not going to use the simplified DSL. As with the `raise` `Klass.new('message')`, I want to **see** that I am creating a `FactoryBot` instance here. The DSL version is too magical and indirect.